src/utils/stark/stark-curve.ts

```diff
@@ -60,17 +65,28 @@ function hashKeyWithIndex(key: string, index: number): BN {
   );
 }
 
-function grindKey(key: BN, keyValLimit: BN) {
+/*
+ This function receives a key seed and produces an appropriate StarkEx key from a uniform
+ distribution.
+ Although it is possible to define a StarkEx key as a residue between the StarkEx EC order and a
+ random 256bit digest value, the result would be a biased key. In order to prevent this bias, we
+ deterministically search (by applying more hashes, AKA grinding) for a value lower than the largest
+ 256bit multiple of StarkEx EC order.
```

Comment:
nit; Stark EC order is in the order of 252 bits `2^251+17⋅2^192+1` not 256 bits.